File: LinkedList_1.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
	int value;
	struct node* next;
};
struct node* head;

struct node* createNode(int val,struct node* pre)
{
	struct node* m=malloc(sizeof(struct node));
	m->value=val;
	m->next=pre;
	return m;
}

void insertAt(int val,int idx)
{
	if(idx<1)
		return;
	struct node* p=head;
	int i;
	if(idx>1)
	{
		for(i=0;i<idx-2;i++)
		{
			if(p!=NULL)
				p=p->next;
			else
			{
				printf("Index is out of range\n");
				return;
			}
		}
		p->next=createNode(val,p->next);
	}
	else
	{
		head=createNode(val,head);
	}
	printf("Element %d is successfully inserted\n",val);
	return;
}

void display()
{
	struct node* p=head;
	while(p!=NULL)
	{
		printf("%d -> ",p->value);
		p=p->next;
	}
	printf("\n");
}

void main()
{
	int i=0;
	while(i!=3)
	{
		printf("Enter choice\n1. Insert\n2. Display\n3. Exit\ni=");
		scanf("%d",&i);	
		if(i==1)
		{
			int val,idx;
			printf("Enter value and index\n");
			scanf("%d %d",&val,&idx);
			insertAt(val,idx);
		}
		else if(i==2)
			display();
		else
		{	
			printf("Are you sure want to exit?\n1. yes\t2.no\n");
			scanf("%d",&i);
			if(i==1)
				return;
			else
				i=0;
		}
		printf("\n");
	}
}

File: PROGRAMS/Python/LinkedList_1.py
class Node:
    def __init__(self,val=None,nxt=None):
        self.value=val
        self.next=nxt

class LinkedList:
    def __init__(self):
        self.head=Node()
        self.count=0

    def insert(self,val,index=None):
        if index==None:
            p=Node(val,self.head.next)
            self.head.next=p
        else:
            if index<1 or index>self.count+1:
                print("index is Invalid.")
            else:
                p=self.head
                for i in range(index-1):
                    p=p.next
                k=Node(val,p.next)
                p.next=k
        self.count+=1
            
    def printLL(self):
        p=self.head
        while p.next!=None:
            print(p.next.value,end=' ')
            p=p.next
        print()

    def size(self):
        print(self.count)
        return self.count
    

def main():
    L=LinkedList()
    L.insert(1)
    L.insert(2)
    L.insert(3)
    L.insert(4)
    L.insert(5,5)
    L.printLL()
    L.size()

main()
